sdk/webpubsub/Azure.Messaging.WebPubSub/src/JwtBuilder.cs

```diff
@@ -0,0 +1,184 @@
+
+        private byte[] _jwt;
+        private int _jwtLength;
+
+        public JwtBuilder(byte[] key, int size = 512)
+        { // typical JWT is ~300B UTF8
+            _jwt = null;
+            _memoryStream = new MemoryStream(size);
+            _memoryStream.Write(headerSha256, 0, headerSha256.Length);
+            _writer = new Utf8JsonWriter(_memoryStream);
+            _writer.WriteStartObject();
+            _key = key;
+        }
+
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, string value) => _writer.WriteString(utf8Name, value);
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, bool value) => _writer.WriteBoolean(utf8Name, value);
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, long value) => _writer.WriteNumber(utf8Name, value);
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, double value) => _writer.WriteNumber(utf8Name, value);
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, DateTimeOffset value)
+        {
+            AddClaim(utf8Name, value.ToUnixTimeSeconds());
+        }
+        public void AddClaim(ReadOnlySpan<byte> utf8Name, string[] value)
+        {
+            _writer.WriteStartArray(utf8Name);
+            foreach (var item in value)
+            {
+                _writer.WriteStringValue(item);
+            }
+            _writer.WriteEndArray();
+        }
+
+        public void AddClaim(string name, string value) => _writer.WriteString(name, value);
+        public void AddClaim(string name, bool value) => _writer.WriteBoolean(name, value);
+        public void AddClaim(string name, long value) => _writer.WriteNumber(name, value);
+        public void AddClaim(string name, double value) => _writer.WriteNumber(name, value);
+        public void AddClaim(string name, DateTimeOffset value)
+        {
+            AddClaim(name, value.ToUnixTimeSeconds());
+        }
+        public void AddClaim(string name, string[] value)
+        {
+            _writer.WriteStartArray(name);
+            foreach (var item in value)
+            {
+                _writer.WriteStringValue(item);
+            }
+            _writer.WriteEndArray();
+        }
+
+        /// <summary>
+        /// Returns number of ASCII characters of the JTW. The actual token can be retrieved using Build or WriteTo
+        /// </summary>
+        /// <returns></returns>
+        public int End()
+        {
+            if (_writer == null) return _jwtLength; // writer is set to null after token is formatted.
+            if (isDisposed) throw new ObjectDisposedException(nameof(JwtBuilder));
+
+            _writer.WriteEndObject();
+            _writer.Flush();
+
+            Debug.Assert(_memoryStream.GetType() == typeof(MemoryStream));
+            int payloadLength = (int)_writer.BytesCommitted; // writer is wrrapping MemoryStream, and so the length will never overflow int.
+
+            int payloadIndex = headerSha256.Length;
+            int maxBufferLength =
```

Comment:
Any concerns about /this/ computation overflowing?

Reply:
It would be very hard to make it overflow, but I changed the code to do checked additions.